### PR TITLE
REF: Use _maybe_cast_indexer and dont override Categorical.get_loc

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -861,6 +861,8 @@ def __getitem__(self, key):
 
                 return result
             except InvalidIndexError:
+                if not isinstance(self.index, MultiIndex):
+                    raise
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]: Is this related to the rest of the PR? Is there a test that otherwise fails?

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good, just a question about Series getitem

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: woops, not related, thought i had reverted this

**Comment by jreback** [comment]
great!

**Pull request merged**